File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len = nums.size();
      
      unordered_map<int, int> mpp;
      
      for(int i=0 ;i<len; i++){
        if(mpp[target-nums[i]] != 0) return {mpp[target-nums[i]]-1, i}; 
        mpp[nums[i]] = i+1;
      }
      return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap <>();
        for(int i=0; i<nums.length; i++){
            if(mp.containsKey(target-nums[i])){
                
                int[] result = { mp.get(target-nums[i]), i };
                return result;
            }
    
            mp.put(nums[i], i);
        }
        return null;
    }
}